src/lib.rs

```diff
@@ -89,6 +89,9 @@
 //! key-value database, where the keys consist of these server-side identifiers
 //! for each client, and the values consist of their corresponding password
 //! files, to be retrieved upon future login attempts made by the client.
+//! It is your responsibility to ensure that the identifier given by the
+//! client in the initial [RegistrationRequest] is the database key you use
+//! in the final [RegistrationUpload] step.
```

Comment:
```suggestion
//! It is your responsibility to ensure that the identifier used in the initial
//! [RegistrationRequest], typically supplied by the client, is the same as the
//! database key used in the final [RegistrationUpload] step.
```
For the purpose of understanding this, I wanted to highlight the crux of the problem here, that in a typical scenario, the `credential_identifier` is supplied by the user, just to make sure not to give the impression that it's required.

It might be useful to mention REST API here, or lack of state? Maybe it's overkill.
Also my English is not the best, so I'm going to refrain from ruining this great PR even more.

Reply:
This looks good!

The only edit I nits I would propose are to replace:
"used in the initial [RegistrationRequest]" -> "used to form the initial [RegistrationRequest]"
"is the same as" -> "matches"

And I think we don't need to mention REST API / statelessness here -- I'm happy with leaving it just at this additional sentence.

Reply:
Did the copy-editing. Thanks!